src/graphics/reproject-texture.js

```diff
@@ -281,74 +306,97 @@ const generateGGXSamples = (numSamples, specularPower, sourceTotalPixels) => {
 };
 
 // pack float samples data into an rgba8 texture
-const packSamplesTex = (device, name, samples) => {
-    const numSamples = samples.length;
-
-    const w = Math.min(numSamples, 512);
-    const h = Math.ceil(numSamples / w);
-    const data = new Uint8Array(w * h * 4);
-
-    // normalize float data and pack into rgba8
-    let off = 0;
-    for (let i = 0; i < numSamples; ++i) {
-        packFloat32ToRGBA8(samples[i * 4 + 0] * 0.5 + 0.5, data, off + 0);
-        packFloat32ToRGBA8(samples[i * 4 + 1] * 0.5 + 0.5, data, off + 4);
-        packFloat32ToRGBA8(samples[i * 4 + 2] * 0.5 + 0.5, data, off + 8);
-        packFloat32ToRGBA8(samples[i * 4 + 3] / 8, data, off + 12);
-        off += 16;
-    }
-
+const createSamplesTex = (device, name, samples) => {
+    const packedSamples = packSamples(samples);
     return new Texture(device, {
         name: name,
-        width: w,
-        height: h,
+        width: packedSamples.width,
+        height: packedSamples.height,
         mipmaps: false,
         minFilter: FILTER_NEAREST,
         magFilter: FILTER_NEAREST,
-        levels: [data]
+        levels: [packedSamples.data]
     });
 };
```

Comment:
Not sure if its worth the change, just an example of "property shorthand notation" and "object deconstructuring" to read/write less code:
```js
    const {width, height, data} = packSamples(samples);
    return new Texture(device, {
        name,
        width,
        height,
        mipmaps: false,
        minFilter: FILTER_NEAREST,
        magFilter: FILTER_NEAREST,
        levels: [data]
    });
```